2021-02-12 23:10:52,017 INFO: Microblog startup [in /Users/cooper/Desktop/Websites/microblog/app/__init__.py:51]
2021-02-12 23:12:03,115 INFO: Microblog startup [in /Users/cooper/Desktop/Websites/microblog/app/__init__.py:51]
2021-02-12 23:14:41,815 INFO: Microblog startup [in /Users/cooper/Desktop/Websites/microblog/app/__init__.py:51]
2021-02-12 23:17:32,177 INFO: Microblog startup [in /Users/cooper/Desktop/Websites/microblog/app/__init__.py:51]
2021-02-12 23:19:59,820 INFO: Microblog startup [in /Users/cooper/Desktop/Websites/microblog/app/__init__.py:51]
2021-02-12 23:21:13,743 INFO: Microblog startup [in /Users/cooper/Desktop/Websites/microblog/app/__init__.py:51]
2021-02-12 23:21:21,347 INFO: Microblog startup [in /Users/cooper/Desktop/Websites/microblog/app/__init__.py:51]
2021-02-12 23:25:41,502 INFO: Microblog startup [in /Users/cooper/Desktop/Websites/microblog/app/__init__.py:51]
2021-02-12 23:25:48,838 INFO: Microblog startup [in /Users/cooper/Desktop/Websites/microblog/app/__init__.py:51]
2021-02-12 23:27:17,527 INFO: Microblog startup [in /Users/cooper/Desktop/Websites/microblog/app/__init__.py:51]
2021-02-12 23:27:22,986 INFO: Microblog startup [in /Users/cooper/Desktop/Websites/microblog/app/__init__.py:51]
2021-02-12 23:27:33,797 INFO: Microblog startup [in /Users/cooper/Desktop/Websites/microblog/app/__init__.py:51]
2021-02-12 23:27:50,990 INFO: Microblog startup [in /Users/cooper/Desktop/Websites/microblog/app/__init__.py:51]
2021-02-12 23:28:30,350 INFO: Microblog startup [in /Users/cooper/Desktop/Websites/microblog/app/__init__.py:51]
2021-02-12 23:28:37,590 INFO: Microblog startup [in /Users/cooper/Desktop/Websites/microblog/app/__init__.py:51]
2021-02-13 00:48:36,130 INFO: Microblog startup [in /Users/cooper/Desktop/Websites/microblog/app/__init__.py:51]
2021-02-13 00:49:54,570 INFO: Microblog startup [in /Users/cooper/Desktop/Websites/microblog/app/__init__.py:51]
2021-02-13 00:51:01,108 INFO: Microblog startup [in /Users/cooper/Desktop/Websites/microblog/app/__init__.py:51]
2021-02-13 00:56:14,165 INFO: Microblog startup [in /Users/cooper/Desktop/Websites/microblog/app/__init__.py:51]
2021-02-13 00:57:44,951 INFO: Microblog startup [in /Users/cooper/Desktop/Websites/microblog/app/__init__.py:51]
2021-02-13 01:38:56,121 INFO: Microblog startup [in /Users/cooper/Desktop/Websites/microblog/app/__init__.py:51]
2021-02-13 01:41:22,555 INFO: Microblog startup [in /Users/cooper/Desktop/Websites/microblog/app/__init__.py:51]
2021-02-13 01:41:56,916 ERROR: Exception on /user/pooopdawg [GET] [in /Users/cooper/Desktop/Websites/microblog/venv/lib/python3.9/site-packages/flask/app.py:1891]
Traceback (most recent call last):
  File "/Users/cooper/Desktop/Websites/microblog/venv/lib/python3.9/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/cooper/Desktop/Websites/microblog/venv/lib/python3.9/site-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/cooper/Desktop/Websites/microblog/venv/lib/python3.9/site-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/Users/cooper/Desktop/Websites/microblog/venv/lib/python3.9/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/Users/cooper/Desktop/Websites/microblog/venv/lib/python3.9/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/cooper/Desktop/Websites/microblog/venv/lib/python3.9/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/Users/cooper/Desktop/Websites/microblog/venv/lib/python3.9/site-packages/flask_login/utils.py", line 272, in decorated_view
    return func(*args, **kwargs)
  File "/Users/cooper/Desktop/Websites/microblog/app/routes.py", line 89, in user
    return render_template('user.html', user=user, posts=posts)
  File "/Users/cooper/Desktop/Websites/microblog/venv/lib/python3.9/site-packages/flask/templating.py", line 137, in render_template
    return _render(
  File "/Users/cooper/Desktop/Websites/microblog/venv/lib/python3.9/site-packages/flask/templating.py", line 120, in _render
    rv = template.render(context)
  File "/Users/cooper/Desktop/Websites/microblog/venv/lib/python3.9/site-packages/jinja2/environment.py", line 1090, in render
    self.environment.handle_exception()
  File "/Users/cooper/Desktop/Websites/microblog/venv/lib/python3.9/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/Users/cooper/Desktop/Websites/microblog/venv/lib/python3.9/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/Users/cooper/Desktop/Websites/microblog/app/templates/user.html", line 1, in top-level template code
    {% extends "base.html" %}
  File "/Users/cooper/Desktop/Websites/microblog/app/templates/base.html", line 30, in top-level template code
    {% block content %}{% endblock %}
  File "/Users/cooper/Desktop/Websites/microblog/app/templates/user.html", line 16, in block "content"
    {{ form.hidden_tag() }}
  File "/Users/cooper/Desktop/Websites/microblog/venv/lib/python3.9/site-packages/jinja2/environment.py", line 471, in getattr
    return getattr(obj, attribute)
jinja2.exceptions.UndefinedError: 'form' is undefined
2021-02-13 01:42:54,993 ERROR: Exception on /user/cooperchen12 [GET] [in /Users/cooper/Desktop/Websites/microblog/venv/lib/python3.9/site-packages/flask/app.py:1891]
Traceback (most recent call last):
  File "/Users/cooper/Desktop/Websites/microblog/venv/lib/python3.9/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/cooper/Desktop/Websites/microblog/venv/lib/python3.9/site-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/cooper/Desktop/Websites/microblog/venv/lib/python3.9/site-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/Users/cooper/Desktop/Websites/microblog/venv/lib/python3.9/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/Users/cooper/Desktop/Websites/microblog/venv/lib/python3.9/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/cooper/Desktop/Websites/microblog/venv/lib/python3.9/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/Users/cooper/Desktop/Websites/microblog/venv/lib/python3.9/site-packages/flask_login/utils.py", line 272, in decorated_view
    return func(*args, **kwargs)
  File "/Users/cooper/Desktop/Websites/microblog/app/routes.py", line 89, in user
    return render_template('user.html', user=user, posts=posts)
  File "/Users/cooper/Desktop/Websites/microblog/venv/lib/python3.9/site-packages/flask/templating.py", line 137, in render_template
    return _render(
  File "/Users/cooper/Desktop/Websites/microblog/venv/lib/python3.9/site-packages/flask/templating.py", line 120, in _render
    rv = template.render(context)
  File "/Users/cooper/Desktop/Websites/microblog/venv/lib/python3.9/site-packages/jinja2/environment.py", line 1090, in render
    self.environment.handle_exception()
  File "/Users/cooper/Desktop/Websites/microblog/venv/lib/python3.9/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/Users/cooper/Desktop/Websites/microblog/venv/lib/python3.9/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/Users/cooper/Desktop/Websites/microblog/app/templates/user.html", line 1, in top-level template code
    {% extends "base.html" %}
  File "/Users/cooper/Desktop/Websites/microblog/app/templates/base.html", line 30, in top-level template code
    {% block content %}{% endblock %}
  File "/Users/cooper/Desktop/Websites/microblog/app/templates/user.html", line 16, in block "content"
    {{ form.hidden_tag() }}
  File "/Users/cooper/Desktop/Websites/microblog/venv/lib/python3.9/site-packages/jinja2/environment.py", line 471, in getattr
    return getattr(obj, attribute)
jinja2.exceptions.UndefinedError: 'form' is undefined
2021-02-15 21:25:15,375 INFO: Microblog startup [in /Users/cooper/Desktop/Websites/microblog/app/__init__.py:51]
2021-02-15 21:31:14,744 INFO: Microblog startup [in /Users/cooper/Desktop/Websites/microblog/app/__init__.py:51]
